src/dispatch/plugins/dispatch_slack/fields.py

```diff
@@ -594,22 +594,22 @@ def case_type_select(
 
 
 def entity_select(
+    signal_id: int,
     db_session: SessionLocal,
     action_id: str = DefaultActionIds.entity_select,
     block_id: str = DefaultBlockIds.entity_select,
     label="Entities",
-    project_id: int = None,
     **kwargs,
 ):
     """Creates an entity select."""
     entity_options = [
         {"text": entity.value, "value": entity.id}
-        for entity in entity_service.get_all(db_session=db_session, project_id=project_id).all()
+        for entity in entity_service.get_all_by_signal(db_session=db_session, signal_id=signal_id)
     ]
 
     return multi_select_block(
         placeholder="Select Entities",
-        options=entity_options,
+        options=entity_options[:100],  # Limit the entities to the first 100
```

Comment:
Is there a way we can be smarter on which ones we offer as an option?

Reply:
Updated to return the 100 most recent entities. There's probably something clever we can do for common use cases (people likely want to snooze when their username shows up, for example). But, this is probably good for now.